runway/hooks/staticsite/upload_staticsite.py

```diff
@@ -42,37 +46,55 @@ def sync(context, provider, **kwargs):
             The provider instance.
 
     """
-    session = get_session(provider.region)
+    session = context.get_session()
     bucket_name = OutputLookup.handle(kwargs.get('bucket_output_lookup'),
                                       provider=provider,
                                       context=context)
+    build_context = context.hook_data['staticsite']
+    invalidate_cache = False
+
+    extra_files = sync_extra_files(
+        context,
+        bucket_name,
+        kwargs.get('extra_files', []),
+        hash_tracking_parameter=build_context.get('hash_tracking_parameter')
+    )
+
+    if len(extra_files) > 0:
+        invalidate_cache = True
 
-    if context.hook_data['staticsite']['deploy_is_current']:
-        LOGGER.info('staticsite: skipping upload; latest version already '
-                    'deployed')
-        if kwargs.get('cf_disabled', '') == 'true':
-            display_static_website_url(kwargs.get('website_url'), provider, context)
+    if build_context['deploy_is_current']:
+        LOGGER.info('staticsite: skipping upload; latest version already deployed')
     else:
         # Using the awscli for s3 syncing is incredibly suboptimal, but on
         # balance it's probably the most stable/efficient option for syncing
         # the files until https://github.com/boto/boto3/issues/358 is resolved
-        aws_cli(['s3',
-                 'sync',
-                 context.hook_data['staticsite']['app_directory'],
-                 "s3://%s/" % bucket_name,
-                 '--delete'])
-
-        if kwargs.get('cf_disabled', False):
-            display_static_website_url(kwargs.get('website_url'), provider, context)
-        else:
-            distribution = get_distribution_data(context, provider, **kwargs)
-            invalidate_distribution(session, **distribution)
+        sync_args = ['s3',
+                     'sync',
+                     build_context['app_directory'],
+                     "s3://%s/" % bucket_name,
+                     '--delete']
+
+        for extra_file in [f['name'] for f in kwargs.get('extra_files', [])]:
+            sync_args.extend(['--exclude', extra_file])
+
+        aws_cli(sync_args)
+
+        invalidate_cache = True
 
         LOGGER.info("staticsite: sync " "complete")
 
         update_ssm_hash(context, session)
 
+    if kwargs.get('cf_disabled', False):
+        display_static_website_url(kwargs.get('website_url'), provider, context)
+
+    if invalidate_cache:
```

Comment:
Good catch, that should be `elif`.

Reply:
Also, invalidate_cache is initialized to False. There are 2 independent scenarios when the cache should be cleared:

1. Any of the extra files change
2. Any of the static site files change

If any of those are true, then the cache will be invalidated. This makes sure that if both are True, we don't have to calls to invalidate the cache.

Reply:
i think i follow how `invalidate_cache` is being determined now. couldn't you use `elif (extra_files or not build_context['deploy_is_current']):` instead of needing the extra variable so that all the decision logic is in the statement.